### PR TITLE
Add basic UI

### PR DESCRIPTION
**Pull request by Surafels**
Hello 👋👋

Implemented tasks:

- [ ] Create a app.rb file that will serve as console app entry-point. It have methods that do the following:

- List all books.
- List all people.
- Create a person (teacher or student, not a plain Person).
- Create a book.
- Create a rental.
- List all rentals for a given person id.
 

- [ ] Create main.rb define the entry point, this will be a method called main that is invoked at the end of  file. This method should do the following:

-  Present the user with a list of options to perform.
- Let users choose an option.
- If needed, ask for parameters for the option.
- Have a way to quit the app.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  def list_all_people
+    puts 'List of people:'
+    @people.each do |person|
+      if person.is_a?(Student)
+        puts "[Student]  Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      elsif person.is_a?(Teacher)
+        puts "[Teacher] Name: #{person.name}, ID: #{person.id},  Age: #{person.age},
+        "
+      else
+        puts 'Invalid either student or teacher please'
```
Comment (LionRouge1) [inline app.rb:28]: - [ ] Nice work implementing the option to list all people :clap: However, listing all the people after creating persons brake the app (see image). Kindly make sure all the created persons are listed when the option 2 is selected.
Screenshot from 2023-11-25 11-29-30
  
  **HIT:** The error is related to trying to get the **id** from the `Person` class without having a **getter** method for **id**. To address this, you need to add `attr_reader :id` to the `Person` class in the `person.rb` file.

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    book_index = gets.chomp.to_i
+    puts "\nSelect a person from the following list by number (not id)"
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+    print "\nDate: "
+    date = gets.chomp
+    @rentals << Rental.new(date, @books[book_index], @people[person_index])
+    puts 'Rental created successfully'
```
Comment (LionRouge1) [inline app.rb]: - [ ] Aside from the **undefined method `id`** mentioned in previous comment, I also encountered  another error when creating a rental (see image). Kindly check it.
Screenshot from 2023-11-25 11-52-10

  **HIT:** The current error appears to be related to the misplacement of arguments when creating a new Rental instance. Kindly make sure you place the argument as in the the `rental.rb` file.
```rb
@rentals << Rental.new(@people[person_index], @books[book_index], date)
```

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @Surafels, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Created `app.rb` file that will serve as their console app entry-point :heavy_check_mark: 
- Have a `main.rb` file for the entry point :heavy_check_mark: 
- App behaves the same as the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ)  provided :x:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      return
+    end
+    rentals = @rentals.select { |r| r.person == person }
+    if rentals.empty?
+      puts "#{person.name} has not done any  rentals"
+    else
+      puts 'Rentals for: '
+      rentals.each { |rental| puts "Date: #{rental.date}, Book '#{rental.book.title}' by #{rental.book.author}" }
+    end
+  end
```
Comment (franclobo) [inline app.rb:114]: - [ ] Well done completing the last required changes to correct the create rentals function. When creating a rental it was created successfully, but when I tried to list, it didn't work. Please 🙏🏽 make sure that you are able to list the rentals. 👇🏽 

image

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @Surafels, 

# Required Changes ♻️

istockphoto-1079018918-612x612

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- No linter errors ✔️ 
- Correct Git flow ✔️ 
- Good UI ✔️ 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi Team,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Followed Gitflow 💯
- [ ]  Descriptive PR⭐
- [ ] No Linters error ✅
- [ ]  Kudos for working on UI🤩
- [ ] All requested changes applied 🚀


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**